File: bloodbank.py
class BloodBank:
    def __init__(self):
        self.blood_inventory = {
            "A+": 10,
            "B+": 15,
            "AB+": 5,
            "O+": 20,
            "A-": 8,
            "B-": 12,
            "AB-": 3,
            "O-": 10,
        }

    def check_blood_availability(self, blood_type):
        return self.blood_inventory.get(blood_type, 0)

    def donate_blood(self, blood_type, amount):
        if blood_type in self.blood_inventory:
            self.blood_inventory[blood_type] += amount
        else:
            self.blood_inventory[blood_type] = amount

    def distribute_blood(self, blood_type, amount):
        if blood_type in self.blood_inventory:
            if self.blood_inventory[blood_type] >= amount:
                self.blood_inventory[blood_type] -= amount
                return True
            else:
                return False
        else:
            return False

def main():
    blood_bank = BloodBank()

    while True:
        print("Blood Bank Application")
        print("1. Check Blood Availability")
        print("2. Donate Blood")
        print("3. Distribute Blood")
        print("4. Exit")
        
        choice = input("Enter your choice: ")

        if choice == "1":
            blood_type = input("Enter blood type: ")
            availability = blood_bank.check_blood_availability(blood_type)
            print(f"Available units of {blood_type}: {availability}")
        
        elif choice == "2":
            blood_type = input("Enter blood type: ")
            amount = int(input("Enter amount to donate: "))
            blood_bank.donate_blood(blood_type, amount)
            print("Thank you for donating!")
        
        elif choice == "3":
            blood_type = input("Enter blood type: ")
            amount = int(input("Enter amount to distribute: "))
            if blood_bank.distribute_blood(blood_type, amount):
                print("Blood distributed successfully.")
            else:
                print("Insufficient blood units.")
        
        elif choice == "4":
            print("Exiting the application.")
            break
        
        else:
            print("Invalid choice. Please select a valid option.")

if __name__ == "__main__":
    main()

File: bloodbank.js
document.addEventListener("DOMContentLoaded", function () {
    const contentDiv = document.getElementById("content");

    // Function to display content in the "content" div
    function displayContent(content) {
        contentDiv.innerHTML = content;
    }

    // Function to send AJAX requests and handle responses
    function sendAjaxRequest(url, method, data, successCallback, errorCallback) {
        fetch(url, {
            method: method,
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(data),
        })
            .then((response) => {
                if (!response.ok) {
                    throw new Error("Network response was not ok");
                }
                return response.json();
            })
            .then(successCallback)
            .catch(errorCallback);
    }

    // Event listener for "Check Blood Availability" button
    document.getElementById("checkBtn").addEventListener("click", () => {
        const content = `
            <h2>Check Blood Availability</h2>
            <label for="bloodType">Enter blood type:</label>
            <input type="text" id="bloodType">
            <button id="checkAvailabilityBtn">Check Availability</button>
            <div id="availabilityResult"></div>
        `;

        displayContent(content);

        // Event listener for "Check Availability" button
        document.getElementById("checkAvailabilityBtn").addEventListener("click", () => {
            const bloodType = document.getElementById("bloodType").value;
            sendAjaxRequest(
                "/check_availability",
                "POST",
                { "blood_type": bloodType },
                (data) => {
                    const availabilityResult = document.getElementById("availabilityResult");
                    availabilityResult.textContent = `Available units of ${bloodType}: ${data.availability}`;
                },
                (error) => {
                    console.error("Error:", error);
                }
            );
        });
    });

    // Add similar event listeners for "Donate Blood" and "Distribute Blood" buttons
});
